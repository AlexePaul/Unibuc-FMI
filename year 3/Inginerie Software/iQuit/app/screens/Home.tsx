import { StyleSheet, Text, View, Button } from "react-native";
import React, { useEffect, useState } from "react";
import { FIREBASE_AUTH, FIRESTORE_DB } from "../../FirebaseConfig";
import { useNavigation } from "@react-navigation/native";
import { StackActions } from "@react-navigation/native";
import { doc, setDoc, getDoc } from "firebase/firestore";
import Dashboard from "./Dashboard";
import DirectQuit from "./DirectQuit";

const Home = () => {
  const [userData, setUserData] = useState(null);
  const [shouldRedirect, setShouldRedirect] = useState(false);

  const navigation = useNavigation();

  const handleSingOut = () => {
    FIREBASE_AUTH.signOut()
      .then(() => {
        console.log("Logged out!");
        navigation.dispatch(StackActions.replace("Landing"));
      })
      .catch((error) => {
        console.log(error);
      });
  };

  const handleDropdownSelect = (index, value) => {
    console.log(`Selected option ${index}: ${value}`);
    // Add your logic here to handle the selected item
  };

  const getUserDataFromFirestore = async (uid) => {
    try {
      const userDocRef = doc(FIRESTORE_DB, "users", uid);
      const userDocSnap = await getDoc(userDocRef);

      if (userDocSnap.exists()) {
        return userDocSnap.data();
      } else {
        console.log("User document does not exist");
        return null;
      }
    } catch (error) {
      console.error("Error getting user data from Firestore:", error);
      return null;
    }
  };

  useEffect(() => {
    const fetchUserData = async () => {
      try {
        const uid = FIREBASE_AUTH.currentUser?.uid;
        console.log(uid);
        if (uid) {
          const userFirestoreData = await getUserDataFromFirestore(uid);
          setUserData(userFirestoreData);
          if (userFirestoreData.direct) {
            setShouldRedirect(true);
          }
          else if (
            userFirestoreData.amount ||
            userFirestoreData.price ||
            userFirestoreData.direct ||
            userFirestoreData.startPlan
          ) {
          } else {
            navigation.reset({
              index: 0,
              routes: [{ name: "PostRegistration" as never }],
            });
          }
        }
      } catch (error) {
        console.error("Error fetching user data:", error);
      }
    };

    fetchUserData();
  }, []);

  if (shouldRedirect) {
    return (
      <View style={styles.container}>
        <DirectQuit />
      </View>
    );
  }
  else {
    return (
      <View>
        <Dashboard />
      </View>
    );
  }
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
