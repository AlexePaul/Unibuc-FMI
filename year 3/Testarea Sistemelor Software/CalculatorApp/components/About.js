import React from 'react';
import { View, Text, StyleSheet, Linking } from 'react-native';

const About = () => {
    return (
        <View style={styles.container}>
            <Text style={styles.text}> App created by zero7zero5 on GitHub </Text>
            <Text
                style={[styles.link]}
                onPress={() => Linking.openURL("https://github.com/zero7zero5/React-Native-Calculator-App")}>
                github.com/zero7zero5/React-Native-Calculator-App
            </Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    text: {
        fontSize: 16,
        margin: 10,
    },
    link: {
        color: 'blue',
        textDecorationLine: 'underline',
    },
});

export default About;