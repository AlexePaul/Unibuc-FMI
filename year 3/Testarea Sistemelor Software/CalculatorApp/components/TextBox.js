import React from "react";
import { StyleSheet, TextInput } from "react-native";

const TextBox = ({
    value,
    onChangeText,
}) => {
    return (
        <TextInput
            maxLength={10}
            testID="text-input"
            value={value}
            keyboardType="number-pad"
            showSoftInputOnFocus={false}
            onChangeText={onChangeText}
            style={styles.input}
        />
    );
};

const styles = StyleSheet.create({
    input: {
        width: "100%",
        height: "30%",
        fontSize: 75,
        textAlign: "right",
        color: "black",
        fontWeight: "300",
    },
});

export default TextBox;
