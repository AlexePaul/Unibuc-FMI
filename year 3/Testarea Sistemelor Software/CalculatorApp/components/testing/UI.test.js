import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import Home from "../Home";
import Button from "../Button";
import TextBox from "../TextBox";
import About from "../About";

describe("UI Testing Suite", () => {
  test("Text box component matches snapshot", () => {
    const tree = renderer.create(<TextBox> 1234 </TextBox>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Button component matches snapshot", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Home screen matches snapshot", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Home screen renders everything", () => {
    const { getByText, getByTestId } = render(<Home />);

    const textBox = getByTestId("text-input");
    expect(textBox).toBeDefined();

    const clearButton = getByText("C");
    expect(clearButton).toBeDefined();

    const ACButton = getByText("AC");
    expect(ACButton).toBeDefined();

    const percentageButton = getByText("%");
    expect(percentageButton).toBeDefined();

    const divideButton = getByText("/");
    expect(divideButton).toBeDefined();

    const multiplyButton = getByText("x");
    expect(multiplyButton).toBeDefined();

    const subtractButton = getByText("-");
    expect(subtractButton).toBeDefined();

    const addButton = getByText("+");
    expect(addButton).toBeDefined();

    const equalsButton = getByText("=");
    expect(equalsButton).toBeDefined();

    const decimalButton = getByText(".");
    expect(decimalButton).toBeDefined();

    const zeroButton = getByText("0");
    expect(zeroButton).toBeDefined();

    const oneButton = getByText("1");
    expect(oneButton).toBeDefined();

    const twoButton = getByText("2");
    expect(twoButton).toBeDefined();

    const threeButton = getByText("3");
    expect(threeButton).toBeDefined();

    const fourButton = getByText("4");
    expect(fourButton).toBeDefined();

    const fiveButton = getByText("5");
    expect(fiveButton).toBeDefined();

    const sixButton = getByText("6");
    expect(sixButton).toBeDefined();

    const sevenButton = getByText("7");
    expect(sevenButton).toBeDefined();

    const eightButton = getByText("8");
    expect(eightButton).toBeDefined();

    const nineButton = getByText("9");
    expect(nineButton).toBeDefined();

    fireEvent.press(oneButton);
    fireEvent.press(twoButton);
    fireEvent.press(threeButton);
    fireEvent.press(fourButton);

    expect(textBox.props.value).toEqual("1234");
  });

  test("About screen matches snapshot", () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
