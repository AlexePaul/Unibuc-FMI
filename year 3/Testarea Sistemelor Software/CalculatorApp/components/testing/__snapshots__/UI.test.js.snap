// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Testing Suite About screen matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "margin": 10,
      }
    }
  >
     App created by zero7zero5 on GitHub 
  </Text>
  <Text
    onPress={[Function]}
    style={
      [
        {
          "color": "blue",
          "textDecorationLine": "underline",
        },
      ]
    }
  >
    github.com/zero7zero5/React-Native-Calculator-App
  </Text>
</View>
`;

exports[`UI Testing Suite Button component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "gold",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "width": 80,
        },
        {
          "backgroundColor": "#cccccc",
          "width": 80,
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 40,
          "fontWeight": "300",
        }
      }
    />
  </View>
</View>
`;

exports[`UI Testing Suite Home screen matches snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "padding": 20,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <TextInput
    keyboardType="number-pad"
    maxLength={10}
    onChangeText={[Function]}
    showSoftInputOnFocus={false}
    style={
      {
        "color": "black",
        "fontSize": 75,
        "fontWeight": "300",
        "height": "30%",
        "textAlign": "right",
        "width": "100%",
      }
    }
    testID="text-input"
    value=""
  />
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "gold",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            AC
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "gold",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            C
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "gold",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            %
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#ffa31a",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            /
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            7
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            8
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            9
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#ffa31a",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            x
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            4
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            5
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            6
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#ffa31a",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            -
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            1
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            2
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            3
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#ffa31a",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 160,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#cccccc",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            .
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "gold",
                "borderRadius": 40,
                "height": 80,
                "justifyContent": "center",
                "width": 80,
              },
              {
                "backgroundColor": "#ffa31a",
                "width": 80,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontSize": 40,
                "fontWeight": "300",
              }
            }
          >
            =
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`UI Testing Suite Text box component matches snapshot 1`] = `
<TextInput
  keyboardType="number-pad"
  maxLength={10}
  showSoftInputOnFocus={false}
  style={
    {
      "color": "black",
      "fontSize": 75,
      "fontWeight": "300",
      "height": "30%",
      "textAlign": "right",
      "width": "100%",
    }
  }
  testID="text-input"
/>
`;
